### PR TITLE
Feat/8 email

### LINKED ISSUE #8: [FEAT] Email SMTP 서비스 구현
## ✨작성할 기능에 대한 설명을 적어주세요
```
로직
이메일 입력 -> discord trigger -> 승인 거절 버튼 -> 승인 시 백엔드 회원가입 서비스로직 실행시켜서 회원가입 시켜줌 -> 해당 이메일로 랜덤난수8자리 전송되야함
```

<br><br>

## ✅ 해야 할 작업들을 작성해주세요
- [x] SMTP 서비스로직구현
- [x] discord와 트리거 설정

<br><br>

## 🦋 연관된 이슈
-

### PR DESCRIPTION
**Pull request by minwoo1999**
## ✨ 요약
```
로직
이메일 입력  -> ADMIN이 승인 시 백엔드 회원가입 서비스로직 실행시켜서 회원가입 시켜줌 -> 해당 이메일로 랜덤난수8자리 전송-> DiscordWebhook을 통해서 회원가입 알림설정

SMTP 서비스로직구현
discord와 트리거 설정
```

<br><br>

## 😎 해결한 이슈
- close #8 

<br><br>

### CODE REVIEW

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [59.29%]|:green_apple:|
|:-|:-:|:-:|
|[MemberDetails.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetails.java)|100%|:green_apple:|
|[MailPurpose.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2FMailPurpose.java)|100%|:green_apple:|
|[GeneRateRandomPassword.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Futils%2FGeneRateRandomPassword.java)|95.95%|:green_apple:|
|[MailSenderServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Fapplication%2Fimpl%2FMailSenderServiceImpl.java)|85.57%|:green_apple:|
|[Member.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fdomain%2FMember.java)|82.57%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fimpl%2FMemberServiceImpl.java)|59.57%|:green_apple:|
|[MemberController.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fpresentation%2FMemberController.java)|55.77%|:green_apple:|
|[JwtAuthorizationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthorizationFilter.java)|50.7%|:green_apple:|
|[MailRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Fapplication%2Fdto%2FMailRequestDto.java)|34.48%|:green_apple:|
|[MailSenderController.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Fpresentation%2FMailSenderController.java)|25.64%|:x:|
|[MailSenderApi.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Finfra%2FMailSenderApi.java)|16.98%|:x:|
|[WebhookService.java](https://github.com/rework-kr/REWORK-SERVER/blob/c2836de51a2bb7faae4807bc240a9edb1661a43f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdiscord%2FWebhookService.java)|10%|:x:|

|Total Project Coverage|60.94%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/example/rework/global/GlobalExceptionHandler.java
Comment (OneK-2) [inline src/main/java/com/example/rework/global/GlobalExceptionHandler.java:1]: ResponseEntity< CommonResponse > 이렇게 작성한거랑, ResponseEntity<?> 이렇게 작성한거 의도를 설명해주실수 있나요??

**Review by OneK-2 — APPROVED** [review]
LGTM~
어제 approve까지 한 줄 알았는데 늦어져 죄송쓰..

File: src/main/java/com/example/rework/global/GlobalExceptionHandler.java
Comment (minwoo1999) [reply src/main/java/com/example/rework/global/GlobalExceptionHandler.java:1]: 제네릭 "?" 타입으로 받게되면 동적으로 타입을 받을 수 있다는 장점이 있는 걸로 알고, CommonResponse로 지정하면 해당하는 타입으로만 받을 수 있다는 것이 장점이라고 알고 있는데 유동적으로 받으려면 "?" 타입으로 받고 정적으로 받으려면 CommonResponse타입으로 받는 것이 더 좋은 것 같네요.. 이부분은 저희끼리 컨벤션 맞춰보면 좋을 것 같아요!

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [59.29%]|:green_apple:|
|:-|:-:|:-:|
|[MemberDetails.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetails.java)|100%|:green_apple:|
|[MailPurpose.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2FMailPurpose.java)|100%|:green_apple:|
|[GeneRateRandomPassword.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Futils%2FGeneRateRandomPassword.java)|95.95%|:green_apple:|
|[MailSenderServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Fapplication%2Fimpl%2FMailSenderServiceImpl.java)|85.57%|:green_apple:|
|[Member.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fdomain%2FMember.java)|82.57%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fimpl%2FMemberServiceImpl.java)|59.57%|:green_apple:|
|[MemberController.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fpresentation%2FMemberController.java)|55.77%|:green_apple:|
|[JwtAuthorizationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthorizationFilter.java)|50.7%|:green_apple:|
|[MailRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Fapplication%2Fdto%2FMailRequestDto.java)|34.48%|:green_apple:|
|[MailSenderController.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Fpresentation%2FMailSenderController.java)|25.64%|:x:|
|[MailSenderApi.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmail%2Finfra%2FMailSenderApi.java)|16.98%|:x:|
|[WebhookService.java](https://github.com/rework-kr/REWORK-SERVER/blob/a273d9ad1aaa9ff40b4fa40381f7e0ed78683ae0/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdiscord%2FWebhookService.java)|10%|:x:|

|Total Project Coverage|60.94%|:green_apple:|
|:-|:-:|:-:|

**Comment by minwoo1999** [comment]
> LGTM~ 어제 approve까지 한 줄 알았는데 늦어져 죄송쓰..

아이고 아니예요 형님ㅋㅋ

**Pull request merged**